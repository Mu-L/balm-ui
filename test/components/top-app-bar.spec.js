import { shallowMount, mount } from '@vue/test-utils';
import UiTopAppBar from '@/components/top-app-bar/top-app-bar.vue';

const CONTENT_ID = 'content-main';

beforeAll(() => {
  const mock_content = document.createElement('div');
  mock_content.id = CONTENT_ID;
  document.body.appendChild(mock_content);
});

describe('<ui-top-app-bar>', () => {
  let methods = {
    init: jest.fn()
  };

  it('renders a default top app bar', () => {
    const icon = 'menu';
    const wrapper = mount(UiTopAppBar, {
      propsData: {
        contentSelector: '#' + CONTENT_ID
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('mdc-top-app-bar');
    expect(wrapper.vm.type).toBe(0);
    expect(wrapper.vm.title).toBe('');
    expect(wrapper.vm.navIcon).toBe(icon);
    expect(wrapper.vm.navId).toBeNull();
  });

  it('renders a top app bar with title', () => {
    const icon = 'menu';
    const title = 'title';
    const wrapper = mount(UiTopAppBar, {
      propsData: {
        contentSelector: '#' + CONTENT_ID,
        title
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.vm.title).toBe(title);
  });

  it('renders a nonRegular top app bar', () => {
    const icon = 'close';
    const type = -1;
    const wrapper = mount(UiTopAppBar, {
      propsData: {
        contentSelector: '#' + CONTENT_ID,
        type
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.findComponent({ name: 'mdc-icon-button' }).text()).toBe(
      icon
    );
  });

  it('renders a fixed top app bar', () => {
    const type = 1;
    const wrapper = mount(UiTopAppBar, {
      propsData: {
        contentSelector: '#' + CONTENT_ID,
        type
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('mdc-top-app-bar--fixed');
  });

  it('renders a dense top app bar', () => {
    const type = 2;
    const wrapper = mount(UiTopAppBar, {
      propsData: {
        contentSelector: '#' + CONTENT_ID,
        type
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('mdc-top-app-bar--dense');
  });

  it('renders a prominent top app bar', () => {
    const type = 3;
    const wrapper = mount(UiTopAppBar, {
      propsData: {
        contentSelector: '#' + CONTENT_ID,
        type
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('mdc-top-app-bar--prominent');
  });

  it('renders a prominent and dense top app bar', () => {
    const type = 4;
    const wrapper = mount(UiTopAppBar, {
      propsData: {
        contentSelector: '#' + CONTENT_ID,
        type
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('mdc-top-app-bar--dense');
    expect(wrapper.classes()).toContain('mdc-top-app-bar--prominent');
  });

  it('renders a short top app bar', () => {
    const type = 5;
    const wrapper = mount(UiTopAppBar, {
      propsData: {
        contentSelector: '#' + CONTENT_ID,
        type
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('mdc-top-app-bar--short');
  });

  it('renders a short collapesed top app bar', () => {
    const type = 6;
    const wrapper = mount(UiTopAppBar, {
      propsData: {
        contentSelector: '#' + CONTENT_ID,
        type
      }
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('mdc-top-app-bar--short');
    expect(wrapper.classes()).toContain('mdc-top-app-bar--short-collapsed');
  });

  // setMethod is deprecated but this is the only way to test the watch method?
  it('calls init method when the type changes', async () => {
    const type_new = 2;

    const wrapper = mount(UiTopAppBar, {
      propsData: {
        contentSelector: '#' + CONTENT_ID
      },
      methods
    });

    await wrapper.setProps({ type: type_new });

    expect(methods.init).toHaveBeenCalled();
  });

  test('init method destories exisited MDCTopAppBar when init', async () => {
    const type_new = 2;

    const wrapper = mount(UiTopAppBar, {
      propsData: {
        contentSelector: '#' + CONTENT_ID
      }
    });

    await wrapper.setProps({ type: type_new });
  });
});
