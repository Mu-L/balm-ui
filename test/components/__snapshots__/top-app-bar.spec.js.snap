// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ui-top-app-bar> renders a default top app bar 1`] = `
<header
  class="mdc-top-app-bar"
>
  <div
    class="mdc-top-app-bar__row"
  >
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
    >
      <span
        class="mdc-top-app-bar__brand"
      >
        <button
          class="mdc-icon-button material-icons mdc-top-app-bar__navigation-icon"
          type="button"
        >
          menu
        </button>
      </span>
       
      <span
        class="mdc-top-app-bar__title"
      >
        
      </span>
    </section>
     
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
      role="toolbar"
    />
  </div>
</header>
`;

exports[`<ui-top-app-bar> renders a dense top app bar 1`] = `
<header
  class="mdc-top-app-bar mdc-top-app-bar--dense"
>
  <div
    class="mdc-top-app-bar__row"
  >
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
    >
      <span
        class="mdc-top-app-bar__brand"
      >
        <button
          class="mdc-icon-button material-icons mdc-top-app-bar__navigation-icon"
          type="button"
        >
          menu
        </button>
      </span>
       
      <span
        class="mdc-top-app-bar__title"
      >
        
      </span>
    </section>
     
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
      role="toolbar"
    />
  </div>
</header>
`;

exports[`<ui-top-app-bar> renders a fixed top app bar 1`] = `
<header
  class="mdc-top-app-bar mdc-top-app-bar--fixed"
>
  <div
    class="mdc-top-app-bar__row"
  >
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
    >
      <span
        class="mdc-top-app-bar__brand"
      >
        <button
          class="mdc-icon-button material-icons mdc-top-app-bar__navigation-icon"
          type="button"
        >
          menu
        </button>
      </span>
       
      <span
        class="mdc-top-app-bar__title"
      >
        
      </span>
    </section>
     
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
      role="toolbar"
    />
  </div>
</header>
`;

exports[`<ui-top-app-bar> renders a nonRegular top app bar 1`] = `
<header
  class="mdc-top-app-bar"
>
  <div
    class="mdc-top-app-bar__row"
  >
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
    >
      <span
        class="mdc-top-app-bar__brand"
      >
        <button
          class="mdc-icon-button material-icons mdc-top-app-bar__navigation-icon"
          type="button"
        >
          close
        </button>
      </span>
       
      <span
        class="mdc-top-app-bar__title"
      >
        
      </span>
    </section>
     
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
      role="toolbar"
    />
  </div>
</header>
`;

exports[`<ui-top-app-bar> renders a prominent and dense top app bar 1`] = `
<header
  class="mdc-top-app-bar mdc-top-app-bar--dense mdc-top-app-bar--prominent"
>
  <div
    class="mdc-top-app-bar__row"
  >
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
    >
      <span
        class="mdc-top-app-bar__brand"
      >
        <button
          class="mdc-icon-button material-icons mdc-top-app-bar__navigation-icon"
          type="button"
        >
          menu
        </button>
      </span>
       
      <span
        class="mdc-top-app-bar__title"
      >
        
      </span>
    </section>
     
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
      role="toolbar"
    />
  </div>
</header>
`;

exports[`<ui-top-app-bar> renders a prominent top app bar 1`] = `
<header
  class="mdc-top-app-bar mdc-top-app-bar--prominent"
>
  <div
    class="mdc-top-app-bar__row"
  >
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
    >
      <span
        class="mdc-top-app-bar__brand"
      >
        <button
          class="mdc-icon-button material-icons mdc-top-app-bar__navigation-icon"
          type="button"
        >
          menu
        </button>
      </span>
       
      <span
        class="mdc-top-app-bar__title"
      >
        
      </span>
    </section>
     
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
      role="toolbar"
    />
  </div>
</header>
`;

exports[`<ui-top-app-bar> renders a short collapesed top app bar 1`] = `
<header
  class="mdc-top-app-bar mdc-top-app-bar--short mdc-top-app-bar--short-collapsed"
>
  <div
    class="mdc-top-app-bar__row"
  >
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
    >
      <span
        class="mdc-top-app-bar__brand"
      >
        <button
          class="mdc-icon-button material-icons mdc-top-app-bar__navigation-icon"
          type="button"
        >
          menu
        </button>
      </span>
       
      <span
        class="mdc-top-app-bar__title"
      >
        
      </span>
    </section>
     
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
      role="toolbar"
    />
  </div>
</header>
`;

exports[`<ui-top-app-bar> renders a short top app bar 1`] = `
<header
  class="mdc-top-app-bar mdc-top-app-bar--short"
>
  <div
    class="mdc-top-app-bar__row"
  >
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
    >
      <span
        class="mdc-top-app-bar__brand"
      >
        <button
          class="mdc-icon-button material-icons mdc-top-app-bar__navigation-icon"
          type="button"
        >
          menu
        </button>
      </span>
       
      <span
        class="mdc-top-app-bar__title"
      >
        
      </span>
    </section>
     
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
      role="toolbar"
    />
  </div>
</header>
`;

exports[`<ui-top-app-bar> renders a top app bar with title 1`] = `
<header
  class="mdc-top-app-bar"
>
  <div
    class="mdc-top-app-bar__row"
  >
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
    >
      <span
        class="mdc-top-app-bar__brand"
      >
        <button
          class="mdc-icon-button material-icons mdc-top-app-bar__navigation-icon"
          type="button"
        >
          menu
        </button>
      </span>
       
      <span
        class="mdc-top-app-bar__title"
      >
        title
      </span>
    </section>
     
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-end"
      role="toolbar"
    />
  </div>
</header>
`;
